### DISCUSSION TITLE
A real language server for Kotlin?

### DISCUSSION THREAD

**Discussion by spaceunifyfifty** [zed-industries/zed#10913] [Feature Requests]
Currently the [Kotlin extension](https://github.com/zed-extensions/zed-kotlin) only provides syntax highlighting, there is no linting, documentation hints or autocomplete suggestions. 

What would it take to get a proper language server for Kotlin into Zed? There is an analogous project for vscode at https://github.com/fwcd/vscode-kotlin (though it has it's own quirks and inadequacies). It uses the language server protocol ([implementation](https://github.com/fwcd/kotlin-language-server)), so could a significant portion of it be ported over to Zed's extension?

How much domain knowledge is realistically required to contribute to this effort? Do we need to find programmers who have years of experience building programming languages from scratch / working on real compilers/parsers for major languages and editors and convince them to help out?

**Comment by spaceunifyfifty** [comment]
some talk about it here https://github.com/zed-extensions/zed-kotlin/issues/1 #10467

**Comment by FalconSlayer51** [comment]
The current kotlin extension for the zed works on macos but it will not work on windows, i have a fix and raised an issue also in the zed kotlin extension repository but currently no one active on that. @spaceunifyfifty

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
